proto/streamlit/proto/NumberInput.proto

```diff
@@ -19,9 +19,23 @@ syntax = "proto3";
 message NumberInput {
   string id = 1;
   string label = 2;
-  float default = 3;
+
+  oneof data {
+    IntNumberInput int_data = 3;
+    FloatNumberInput float_data = 4;
+  }
+
   float min = 5;
   float max = 6;
```

Comment:
Not high priority, but these should be inside IntNumberInput and FloatNumberInput too. Otherwise there will be float-to-int conversion problems with very large numbers.

The only problem is we'd then have to create two new fields to track whether a max/min is set when using integers (since `int` doesn't support `Infinity`). Something like `int_data.has_max` and `int_data.has_min`

Reply:
@tvst Did you mean to say create two new fields to track whether max/min are set to infinity?